AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_HairAdded.py

```diff
@@ -0,0 +1,184 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+class Tests:
+    hair_creation = (
+        "Hair Entity successfully created",
+        "P0: Hair Entity failed to be created")
+    hair_component = (
+        "Entity has a Hair component",
+        "P0: Entity failed to find Hair component")
+    hair_component_removal = (
+        "Hair component removed",
+        "P0: Hair component removal failed")
+    hair_disabled = (
+        "Hair component disabled",
+        "P0: Hair component was not disabled.")
+    actor_component = (
+        "Entity has an Actor component",
+        "P0: Entity did not have an Actor component")
+    hair_enabled = (
+        "Hair component enabled",
+        "P0: Hair component was not enabled.")
+    hair_asset = (
+        "Hair Asset property set",
+        "P0: Hair Asset property was not set as expected")
+    enter_game_mode = (
+        "Entered game mode",
+        "P0: Failed to enter game mode")
+    exit_game_mode = (
+        "Exited game mode",
+        "P0: Couldn't exit game mode")
+    is_visible = (
+        "Entity is visible",
+        "P0: Entity was not visible")
+    is_hidden = (
+        "Entity is hidden",
+        "P0: Entity was not hidden")
+    entity_deleted = (
+        "Entity deleted",
+        "P0: Entity was not deleted")
+    deletion_undo = (
+        "UNDO deletion success",
+        "P0: UNDO deletion failed")
+    deletion_redo = (
+        "REDO deletion success",
+        "P0: REDO deletion failed")
+
+
+def AtomEditorComponents_Hair_AddedToEntity():
+    """
+    Summary:
+    Tests the Hair component can be added to an entity and has the expected functionality.
+
+    Test setup:
+    - Wait for Editor idle loop.
+    - Open the "Base" level.
+
+    Expected Behavior:
+    The component can be added, used in game mode, hidden/shown, deleted, and has accurate required components.
+    Creation and deletion undo/redo should also work.
+
+    Test Steps:
+    1) Create a Hair entity with no components.
+    2) Add a Hair component to Hair entity.
+    3) Remove Hair component.
+    4) UNDO the component removal.
+    5) Verify Hair component not enabled.
+    6) Add Actor component since it is required by the Hair component.
+    7) Verify Hair component is enabled.
+    8) Enter/Exit game mode.
+    9) Test IsHidden.
+    10) Test IsVisible.
+    11) Delete Hair entity.
+    12) UNDO deletion.
+    13) REDO deletion.
+    14) Look for errors or asserts.
+
+    :return: None
+    """
+
+    import azlmbr.legacy.general as general
+    # import azlmbr.bus as bus
```

Comment:
why not remove these commented out lines instead of comment them out?

Reply:
excellent point. removing.